packages/graphql-lowcode/src/modify/addFragmentFields.ts

```diff
@@ -0,0 +1,72 @@
+import { graphql, parse, visit } from 'graphql'
+
+export function addFragmentField(graphqlQuery: string, options: { entity: string, property: string }) {
+  //if fragment does not contain fragment keyword -> returns unchanged query
+  if (!graphqlQuery.indexOf('fragment')) return graphqlQuery
+
+  const ast = parse(graphqlQuery)
+  let finalQuery = graphqlQuery
+
+  let lastFragmentFieldIndexes: number[] = []
+  let firstFragmentFieldIndex: number = -1
+
+  visit(ast, {
+    Field(node) {
+      if (node.name.value === options.entity) {
+        const firstFragmentField = node.selectionSet?.selections[0]
+        const lastFragmentField = node.selectionSet?.selections[node.selectionSet.selections.length - 1]
+
+        //get the indexes where the fragment fields starts and ends in query
+        const startIndex = firstFragmentField?.loc?.start
+        const endIndex = lastFragmentField?.loc?.end
+
+        //if is the first iteration get first get first fragment field for formatting
+        if(firstFragmentFieldIndex == -1) firstFragmentFieldIndex = startIndex ? startIndex : -1
```

Comment:
This `if` can be simplified:
`if(firstFragmentFieldIndex == -1 && startIndex) firstFragmentFieldIndex = startIndex`